resteasy-client-microprofile-base/src/main/java/org/jboss/resteasy/microprofile/client/publisher/SSEPublisher.java

```diff
@@ -0,0 +1,290 @@
+package org.jboss.resteasy.microprofile.client.publisher;
+
+import org.jboss.logging.Logger;
+import org.jboss.resteasy.core.ResteasyContext;
+import org.jboss.resteasy.plugins.providers.sse.SseEventInputImpl;
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+import javax.ws.rs.ext.Providers;
+import javax.ws.rs.sse.InboundSseEvent;
+
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.Map;
+import java.util.Queue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Publisher implementation emitting server-sent-event downstream.
+ *
+ * @param <T> the type of event.
+ */
+@SuppressWarnings("ReactiveStreamsPublisherImplementation")
+public class SSEPublisher<T> implements Publisher<T> {
+
+    private static final Runnable CLEARED = () -> {
+        // sentinel indicating we are done.
+    };
+
+    private final SseEventInputImpl input;
+    private final Type genericType;
+    private final Providers providers;
+    private final ExecutorService executor;
+
+    private static final Logger LOGGER = Logger.getLogger(SSEPublisher.class);
+
+    public SSEPublisher(final Type genericType, final Providers providers, final SseEventInputImpl input, final ExecutorService es) {
+        this.genericType = genericType;
+        this.input = input;
+        this.providers = providers;
+        this.executor = es;
+    }
+
+    @Override
+    public void subscribe(final Subscriber<? super T> downstream) {
+        SSEProcessor<? super T> processor = new SSEProcessor<>(downstream,
+                Integer.getInteger("resteasy.microprofile.sseclient.buffersize", 512));
+        downstream.onSubscribe(processor);
+        pump(processor, input);
+    }
+
+    /**
+     * Reads the server-sent event from the {@code input} and pass them to the processor.
+     * The processor handles the downstream requests and buffer/drop items according to them.
+     *
+     * @param processor the stream
+     * @param input     the SSE input
+     */
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    private void pump(final SSEProcessor processor, final SseEventInputImpl input) {
+        Map<Class<?>, Object> contextDataMap = ResteasyContext.getContextDataMap();
+        executor.execute(() -> {
```

Comment:
should it handle rejection?

Reply:
Do you mean handle executor injection ?

Reply:
if the executor is stopped (for whatever reason), you cannot submit another task. You will receive a `RejectedExecutionException`. What should we do in this case? Log and return a failed publisher? Log and return a `Publisher` just emitting the completion signal?

Reply:
@cescoffier Thanks for pointing out this issue.  I modified to this to create a new thread to run this if there is RejectedExecutionException thrown.